support-workers/src/main/scala/com/gu/stripe/StripeService.scala

```diff
@@ -15,16 +17,37 @@ class StripeService(config: StripeConfig, client: FutureHttpClient, baseUrl: Str
   val wsUrl = baseUrl
   val httpClient: FutureHttpClient = client
 
-  def createCustomer(token: String, currency: Currency): Future[Customer] =
-    postForm[Customer]("customers", Map(
-      "card" -> Seq(token)
-    ), getHeaders(currency))
+  def withCurrency(currency: Currency): WithCurrency = new WithCurrency(currency)
+  class WithCurrency(currency: Currency) {
 
-  private def getHeaders(currency: Currency) =
-    config.version.foldLeft(getAuthorizationHeader(currency)) {
-      case (map, version) => map + ("Stripe-Version" -> version)
-    }
+    def post[A](
+      endpoint: String,
+      data: Map[String, Seq[String]]
+    )(implicit decoder: Decoder[A], errorDecoder: Decoder[StripeError], ctag: ClassTag[A]): Future[A] =
+      StripeService.super.postForm[A](endpoint, data, getHeaders())
+
+    private def getHeaders() =
+      config.version.foldLeft(getAuthorizationHeader()) {
+        case (map, version) => map + ("Stripe-Version" -> version)
+      }
+
+    private def getAuthorizationHeader() =
+      Map("Authorization" -> s"Bearer ${config.forCurrency(Some(currency)).secretKey}")
+
+  }
+}
+
+object CreateCustomer {
+  // https://stripe.com/docs/api/customers/create
+
+  def fromSource(token: String): StripeService#WithCurrency => Future[Customer] =
+    _.post[Customer]("customers", Map(
+      "source" -> Seq(token)
```

Comment:
this was "card" but according to stripe docs card was removed in 2015 version of the API and we should have been using "source" all along! Somehow we got away with it though.

Reply:
Backwards compatibility FTW!